src/Parser/Core/Modules/DamageMitigation/DamageMitigation.js

```diff
@@ -0,0 +1,380 @@
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+import EnemyInstances from 'Parser/Core/Modules/EnemyInstances';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Armor from './Armor';
+import Reduction from './Reduction';
+import AOE_TYPE from './AOE_TYPE';
+import AOE_ABILITIES from './AOE_ABILITIES';
+import { BUFFS, DEBUFFS, PASSIVES, UNKNOWN, AURA_OF_SACRIFICE, DEVOTION_AURA } from './Reductions';
+
+const debug = true;
+
+const ACCURACY_THRESHOLD = 0.05;
+const BUFFER_PERCENT = 0.005;
+
+const SAC_DR_VALUES = [0, 0.1, 0.3];
+const DEV_DR_VALUES = [0.03, 0.06125, 0.2];
+
+class DamageMitigation extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+    enemies: EnemyInstances,
+    combatants: Combatants,
+    armor: Armor,
+  };
+
+  static REDUCTION_CLASS = Reduction;
+
+  buffs = [];
+  debuffs = [];
+  passives = [];
+  unknownReduction = null;
+  auraSacrifice = null;
+  auraDevotion = null
+  mitigated = {};
+  checkSum = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.addReductions();
+  }
+
+  on_finished() {
+    if (!debug) {
+      return;
+    }
+    console.log(this.mitigated);
+    console.log('Total: ' + formatNumber(this.totalMitigated) + ', Checksum: ' + formatNumber(this.checkSum));
+    console.log('Unknown: ' + formatNumber(this.mitigated[this.unknownReduction.id].amount) + ', ' + formatPercentage(this.mitigated[this.unknownReduction.id].amount / this.totalMitigated) + '%');
+  }
+
+  get totalMitigated() {
+    let sum = 0;
+    Object.keys(this.mitigated).forEach(key => {
+      sum += this.mitigated[key].amount;
+    });
+    return sum;
+  }
+
+  get isAccurate() {
+    return Math.abs(this.mitigated[this.unknownReduction.id].amount / this.totalMitigated) < ACCURACY_THRESHOLD;
+  }
+
+  addReductions() {
+    // Buffs
+    this.buffs = BUFFS.map(options => new this.constructor.REDUCTION_CLASS(this, options))
+    .filter(e => e.enabled);
+    // Debuffs
+    this.debuffs = DEBUFFS.map(options => new this.constructor.REDUCTION_CLASS(this, options))
+    .filter(e => e.enabled);
+    // Passives
+    this.passives = PASSIVES.map(options => new this.constructor.REDUCTION_CLASS(this, options))
+    .filter(e => e.enabled);
+    // Unknown
+    this.unknownReduction = new this.constructor.REDUCTION_CLASS(this, UNKNOWN);
+    this.initReduction(this.unknownReduction);
+    // Aura of Sacrifice
+    this.auraSacrifice = new this.constructor.REDUCTION_CLASS(this, AURA_OF_SACRIFICE);
+    this.initReduction(this.auraSacrifice);
+    // Devotion Aura
+    this.auraDevotion = new this.constructor.REDUCTION_CLASS(this, DEVOTION_AURA);
+    this.initReduction(this.auraDevotion);
+  }
+
+  initReduction(reduction) {
+    this.mitigated[reduction.id] = {
+      name: reduction.name,
+      amount: 0,
+    };
+  }
+
+  getMitigation(reduction, event) {
+    // Calculate Dampen Harm reduction. 
+    if (reduction.id === SPELLS.DAMPEN_HARM_TALENT.id) {
```

Comment:
I'd prefer it if this file didn't include effects that are class/spec specific. I believe core shouldn't know about class/spec specific things unless they can be applied to others (e.g. externals).